File: app/src/main/java/com/duy/pascal/backend/exceptions/ParsingException.java
package com.duy.pascal.backend.exceptions;


import com.duy.pascal.backend.linenumber.LineInfo;

public class ParsingException extends Exception {
    private static final long serialVersionUID = -5592261827134726938L;
    public LineInfo line;

    public ParsingException() {

    }

    public ParsingException(LineInfo line, String message) {
        super(message);
        this.line = line;
    }

    public ParsingException(LineInfo line) {
        super();
        this.line = line;
    }

    @Override
    public String toString() {
        return line + ":" + getMessage();
    }
}

File: app/src/main/java/com/duy/pascal/interperter/exceptions/parsing/ParsingException.kt
package com.duy.pascal.interperter.exceptions.parsing

import com.duy.pascal.interperter.linenumber.LineInfo


open class ParsingException : Exception {
    var lineInfo: LineInfo? = null

    constructor(lineInfo: LineInfo?, message: String) : super(message) {
        this.lineInfo = lineInfo
    }

    constructor(lineInfo: LineInfo?) {
        this.lineInfo = lineInfo
    }

    override fun toString(): String {
        return this.lineInfo.toString() + ":" + this.message
    }

    open val canAutoFix
        get() = false
}
